### PR TITLE
Initial support for MacOS

### PR DESCRIPTION
**Pull request by ACTom**

### CODE REVIEW

File: peazip-sources/unit_gwrap.pas [hunk 1]
```diff
@@ -2049,7 +2050,18 @@ procedure TForm_gwrap.FormShow(Sender: TObject);
 begin
 if pldesigned=true then exit;
 getdesk_env(desk_env,caption_build,delimiter);
+
+executable_path:=extractfilepath((paramstr(0))); //valorize application's paths
```
Comment (peazip) [inline peazip-sources/unit_gwrap.pas:2054]: executable_path is passed from unit peach prepare_Form_gwrap;

File: peazip-sources/unit_gwrap.pas [hunk 2]
```diff
@@ -2049,7 +2050,18 @@ procedure TForm_gwrap.FormShow(Sender: TObject);
...
+
+executable_path:=extractfilepath((paramstr(0))); //valorize application's paths
+if ((executable_path='') or (executable_path='..\')) then executable_path:=extractfilepath((paramstr(0)));
+if executable_path<>'' then
+   if executable_path[length(executable_path)]<>directoryseparator then executable_path:=executable_path+directoryseparator;
 setcurrentdir(executable_path);
+{$IFDEF DARWIN}
+   resource_path:=executable_path+'../Resources/';
+{$ELSE}
+   resource_path:=resource_path;
```
Comment (peazip) [inline peazip-sources/unit_gwrap.pas:2062]: resource_path:=executable_path+'res'+DirectorySeparator;

File: peazip-sources/unit_pea.pas [hunk 3]
```diff
@@ -7268,13 +7273,16 @@ procedure readconf_relativeline(nlines:integer; var dummy:ansistring);
...
+   if (confpath='appdata') or (confpath='"appdata"') or (confpath='''appdata''') or (confpath='%appdata%') then confpath:=GetEnvironmentVariable('HOME')+'/.PeaZip/';
    {$ENDIF}
    if not(directoryexists(confpath)) then mkdir(confpath);
    if (confpath[1]='.') and (confpath[2]='.') then confpath:='..'+directoryseparator+confpath; //relative path, needs to be adjusted since pea is in a subfolder of peazip path
    confpath:=expandfilename(confpath);
    if confpath[length(confpath)]<>directoryseparator then confpath:=confpath+directoryseparator;
-   if not(directoryexists(confpath)) then confpath:=executable_path; //if alternative configuration directory does not exist or is not accessible, use res path
-   persistent_source:=confpath+'rnd';
+   if not(directoryexists(confpath)) then confpath:=resource_path; //if alternative configuration directory does not exist or is not accessible, use res path
+   persistent_source:=resource_path+'rnd';
```
Comment (peazip) [inline peazip-sources/unit_pea.pas:7285]: persistent_source: should be in confpath which is unique for each user. If it is necessary, I'll add that modification in a conditional compilation section spèecific to Darwin
   {$IFDEF DARWIN}
   persistent_source:=resource_path+'rnd';
   {$ENDIF}

**Review by peazip — COMMENTED** [review]
So far I've successfully tested list_utils, unit_gwrap, and unit_pea sources, whit a few issues discussed in comments.

File: peazip-sources/peach.pas [hunk 4]
```diff
@@ -57038,6 +57048,11 @@ procedure setbasevars;
 if executable_path<>'' then
    if executable_path[length(executable_path)]<>directoryseparator then executable_path:=executable_path+directoryseparator;
 setcurrentdir(executable_path);
+{$IFDEF DARWIN}
+   resource_path:=executable_path+'../Resources/';
+{$ELSE}
+   resource_path:=resource_path;
```
Comment (peazip) [inline peazip-sources/peach.pas:57054]: Structure is 
(peazip app folder)/ contains peazip binary
(peazip app folder)/res/contains resources, so - in non Darwin build - resource_path needs to be
resource_path:=executable_path+'res'+directoryseparator

File: peazip-sources/peach.pas [hunk 5]
```diff
@@ -49612,7 +49617,11 @@ procedure TForm_peach.mpeautilsClick(Sender: TObject);
    cl:ansistring;
 begin
 P:=tprocessutf8.Create(nil);
-cl:=stringdelim(escapefilename(executable_path,desk_env)+'res'+DirectorySeparator+'pea'+EXEEXT);
+{$IFDEF DARWIN}
+cl:='open ' + executable_path + '../../../pea.app';
+{$ELSE}
+cl:=stringdelim(escapefilename(executable_path,desk_env)+'pea'+EXEEXT);
```
Comment (peazip) [inline peazip-sources/peach.pas:49623]: pea binary is in (peazip)/res subfolder

File: peazip-sources/unit_gwrap.pas (hunk 2 above)
Comment (ACTom) [reply peazip-sources/unit_gwrap.pas:2062]: Sorry, I have did "replace all from 'executable_path+'res'+DirectorySeparator' to 'resource_path'", and no found this has replaced either.

File: peazip-sources/peach.pas (hunk 5 above)
Comment (ACTom) [reply peazip-sources/peach.pas:49623]: In Macos, both peazip.app and pea.app should be in the /Applications directory. Of course, if you don’t want to expose pea.app, you can also put pea.app in Resources or other self-built directories.

File: peazip-sources/unit_gwrap.pas (hunk 2 above)
Comment (peazip) [reply peazip-sources/unit_gwrap.pas:2062]: No problem, it is fixed in the updated sources I published yesterday.

File: peazip-sources/peach.pas (hunk 5 above)
Comment (peazip) [reply peazip-sources/peach.pas:49623]: I understand. In the updated sources published yesterday in the peazip-sources directory I've left unchanged your conditional compilation code for Darwin and changed the code for other platforms.

{$IFDEF DARWIN}
cl:='open ' + executable_path + '../../../pea.app';
{$ELSE}
cl:=stringdelim(escapefilename(resource_path,desk_env)+'pea'+EXEEXT);
{$ENDIF}

If this is OK for Darwin, I'll use the same conditional compilation mechanism for the other instances pea binary is invoked, eg. compose_pea_cl, compose_rfs_cl etc.

**Pull request merged**